### PR TITLE
🐛 Delete OS disk when failed VM gets deleted

### LINKED ISSUE #755: When deleting a VM in failed state, OS disk is not deleted
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
Results in error `Only CreateOption.Attach is supported` if an internal (transient) Azure error occurs, when the VM is deleted and we try to create it again with the same existing OS disk.

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What this PR does / why we need it**: Fixes an error `Only CreateOption.Attach is supported` if an internal (transient) Azure error occurs, when the VM is deleted and we try to create it again with the same existing OS disk.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #755 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
fix: Delete OS disk when failed VM gets deleted
```

### CODE REVIEW

File: controllers/azuremachine_reconciler.go [hunk 1]
```diff
@@ -252,6 +252,13 @@ func (s *azureMachineService) reconcileVirtualMachine(ctx context.Context, nicNa
 			if err != nil {
 				return nil, errors.Wrapf(err, "failed to delete machine")
 			}
+			OSDiskSpec := &disks.Spec{
+				Name: azure.GenerateOSDiskName(s.machineScope.Name()),
+			}
+			err = s.disksSvc.Delete(ctx, OSDiskSpec)
```
Comment (CecileRobertMichon) [inline controllers/azuremachine_reconciler.go:258]: /hold

I probably need to do the same for VMSS

Comment (devigned) [reply controllers/azuremachine_reconciler.go:258]: I don't think you do. We don't specify the disk name with VMSS.

Comment (CecileRobertMichon) [reply controllers/azuremachine_reconciler.go:258]: that's true

/hold cancel

**Pull request merged**